include/exadg/acoustic_conservation_equations/spatial_discretization/operators/operator.h

```diff
@@ -0,0 +1,493 @@
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+    }
+    else
+    {
+      AssertThrow(false, dealii::ExcMessage("Not implemented."));
+    }
+  }
+
+
+  template<bool weight_neighbor, // = false for cell centric loops and boundary loops
+           typename ExteriorFaceIntegratorP,
+           typename ExteriorFaceIntegratorU>
+  void
+  do_face_integral(FaceIntegratorP &         pressure_m,
+                   ExteriorFaceIntegratorP & pressure_p,
+                   FaceIntegratorU &         velocity_m,
+                   ExteriorFaceIntegratorU & velocity_p) const
+  {
+    for(unsigned int q : pressure_m.quadrature_point_indices())
+    {
+      scalar const pm = pressure_m.get_value(q);
+      scalar const pp = pressure_p.get_value(q);
+      vector const um = velocity_m.get_value(q);
+      vector const up = velocity_p.get_value(q);
+      vector const n  = pressure_m.normal_vector(q);
+
+      vector const flux_momentum =
+        kernel.calculate_lax_friedrichs_flux_momentum(um, up, gamma, pm, pp, n);
+
+      scalar const flux_mass = kernel.calculate_lax_friedrichs_flux_mass(pm, pp, tau, um, up, n);
+
+      if(data.formulation == Formulation::Weak)
+      {
+        scalar const flux_momentum_weak = rhocc * flux_momentum * n;
+        vector const flux_mass_weak     = rho_inv * flux_mass * n;
+
+        pressure_m.submit_value(flux_momentum_weak, q);
+        velocity_m.submit_value(flux_mass_weak, q);
+
+        if constexpr(weight_neighbor)
+        {
+          // minus signs since n⁺ = - n⁻
+          pressure_p.submit_value(-flux_momentum_weak, q);
+          velocity_p.submit_value(-flux_mass_weak, q);
+        }
+      }
+      else if(data.formulation == Formulation::Strong)
+      {
+        pressure_m.submit_value(rhocc * (flux_momentum - um) * n, q);
+        velocity_m.submit_value(rho_inv * (flux_mass - pm) * n, q);
+
+        if constexpr(weight_neighbor)
+        {
+          // minus signs since n⁺ = - n⁻
+          pressure_p.submit_value(-rhocc * (flux_momentum - up) * n, q);
+          velocity_p.submit_value(-rho_inv * (flux_mass - pp) * n, q);
+        }
+      }
+      else if(data.formulation == Formulation::SkewSymmetric)
+      {
+        scalar const flux_momentum_weak = rhocc * flux_momentum * n;
+
+        pressure_m.submit_value(flux_momentum_weak, q);
+        velocity_m.submit_value(rho_inv * (flux_mass - pm) * n, q);
+
+        if constexpr(weight_neighbor)
+        {
+          // minus signs since n⁺ = - n⁻
+          pressure_p.submit_value(-flux_momentum_weak, q);
+          velocity_p.submit_value(-rho_inv * (flux_mass - pp) * n, q);
+        }
+      }
+      else
+      {
+        AssertThrow(false, dealii::ExcMessage("Not implemented."));
+      }
```

Comment:
Given the difficulties in evaluating the interior/exterior values in a generic way and for all types of loops (see discussion here https://github.com/exadg/exadg/pull/587/files#r1389376858, isn't it actually this part of the code that can be written generically? @jh66637 In the non-matching case, you do not integrate over the neighboring face, right? Then, writing these lines of code as a kernel function ("kernel" in the meaning of a quadrature point function) would not require templating with `ExteriorFaceIntegratorP` and `ExteriorFaceIntegratorU`, because we are always operating on the "normal" `dealii::FaceIntegrator/FEFaceEvaluation`?

@kronbichler would be great if you could also share your opinion.